### PR TITLE
Suppress progress bars in test and debug fixes #254

### LINKED ISSUE #254: Suppress progress bars in tests
This may be a dual issue - i see a lot of piling up of progress bars when running `npm test` -- but not sure if this is because multiple bars are running at once in tests, or what. But in general i think we should suppress progress bars in tests, does this make sense?

image

### PR DESCRIPTION
**Pull request by tech4GT**
Signed-off-by: tech4GT <varun.gupta1798@gmail.com>
@jywarren please have a look

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -1,165 +1,182 @@
 #!/usr/bin/env node
 
-require('./src/ImageSequencer');
-sequencer = ImageSequencer({ui: false});
-var Spinner = require('ora')
+require("./src/ImageSequencer");
```
Comment (ryzokuken) [inline index.js]: Why change all the quotes?

1. It adds noise to the change.
2. Single quotes are usually preferred anyway.

Comment (tech4GT) [reply index.js]: Oh I get this, this is my ide actually😅

Comment (tech4GT) [reply index.js]: I'll get this

**Comment by jywarren** [comment]
Agreed, though thank you for this!!! Perhaps we can do a run of formatting in its own PR?

**Comment by ryzokuken** [comment]
@jywarren exactly what I had in mind.

**Comment by tech4GT** [comment]
@jywarren so we are doing it separate pr then? So this is fine for now??

**Comment by jywarren** [comment]
Can you indicate the lines which do the progress bar suppression so I can review? Thank you!!!!!

**Comment by tech4GT** [comment]
@jywarren actually all I have done is add an environment variable TEST when running the tests and checked for it before drawing progress bars

File: src/modules/_nomodule/PixelManipulation.js [hunk 2]
```diff
@@ -3,84 +3,84 @@
...
+
         pixels.set(x, y, 0, pixel[0]);
         pixels.set(x, y, 1, pixel[1]);
         pixels.set(x, y, 2, pixel[2]);
         pixels.set(x, y, 3, pixel[3]);
-        
-        if(!options.inBrowser)
-        pace.op()
+
+        if (!options.inBrowser && !process.env.TEST) pace.op();
```
Comment (tech4GT) [inline src/modules/_nomodule/PixelManipulation.js]: @jywarren

File: package.json [hunk 3]
```diff
@@ -1,20 +1,20 @@
...
-  "description": "A modular JavaScript image manipulation library modeled on a storyboard.",
+  "description":
+    "A modular JavaScript image manipulation library modeled on a storyboard.",
   "main": "src/ImageSequencer.js",
   "scripts": {
-    "debug": "node ./index.js -i ./examples/images/monarch.png -s",
-    "test": "tape test/**/*.js test/*.js | tap-spec; browserify test/modules/image-sequencer.js test/modules/chain.js test/modules/replace.js | tape-run --render=\"tap-spec\""
+    "debug":
+      "TEST=true node ./index.js -i ./examples/images/monarch.png -s invert",
+    "test":
```
Comment (tech4GT) [inline package.json]: @jywarren

File: src/modules/_nomodule/PixelManipulation.js [hunk 4]
```diff
@@ -3,84 +3,84 @@
...
     }
-    
+
     // iterate through pixels;
     // TODO: this could possibly be more efficient; see
     // https://github.com/p-v-o-s/infragram-js/blob/master/public/infragram.js#L173-L181
-    
-    if (!options.inBrowser) {
+
+    if (!options.inBrowser && !process.env.TEST) {
```
Comment (tech4GT) [inline src/modules/_nomodule/PixelManipulation.js]: @jywarren

Comment (jywarren) [reply src/modules/_nomodule/PixelManipulation.js]: Perfect, that's all I needed. Merging now!

**Pull request merged**